manager/scheduler/nodeinfo.go

```diff
@@ -16,6 +20,11 @@ type NodeInfo struct {
 	ActiveTasksCountByService map[string]int
 	AvailableResources        api.Resources
 
+	// Maps tasks to resource taken usage: `map[taskID][resourceName]`
+	// This is needed to keep track of the resources where the user specified
+	// an int for a node which advertises a set for the specified resource
+	ThirdPartyResourcesTaken tprTaskStore
```

Comment:
I'm wondering if we still need this now that we have `AssignedThirdPartyResources` on `Task`. `ThirdPartyResourcesTaken[taskID]` seems like it could be equivalent to `Tasks[taskID].AssignedThirdPartyResources`.